src/hooks/hooks.js

```diff
@@ -50,14 +50,16 @@ export function useMemo(currentMemo, currentArgs) {
 }
 /**
  * Apply the redux pattern as a hook
- * @param {(state:any,action:any)=>any} reducer
- * @param {any} initialState
+ * @param {(state:any,action:any)=>any} reducer - State reducer function.
+ * @param {any} initialArg - Optional initial state or payload that is passed
+ *      to the lazy state initializer function.
+ * @param {(initialArg)=>any} init - Optional lazy state initializer function.
  */
-export function useReducer(reducer, initialState) {
+export function useReducer(reducer, initialArg, init) {
     let render = useUpdate();
     return useHook((state = []) => {
         if (!state[1]) {
-            state[0] = initialState;
+            state[0] = init != null ? init(initialArg) : initialArg;
```

Comment:
```suggestion
            state[0] = init !== undefined ? init(initialArg) : initialArg;
```

1. **Should check for `undefined` instead of `null`** 
  Undefined optional arguments are of type `undefined` by default (not `null`), or am I missing something?
2. **Using strict equality check `!==` instead of `!=`**  
  Since we mostly check for the **type** of the `init` argument and not for the value, using strict equality would make more sense here.